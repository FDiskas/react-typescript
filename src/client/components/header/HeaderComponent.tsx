import * as React from 'react';

import * as styles from './HeaderComponent.scss';
import { NavigationComponent } from '../navigation/NavigationComponent';

export interface IHeaderProps {}

export interface IHeaderState {}

export class HeaderComponent extends React.Component<IHeaderProps, IHeaderState> {
    constructor(props: IHeaderProps) {
        super(props);
    }

    render() {
        return (
            <div className={styles.header}>
                Component name: HeaderComponent
                <NavigationComponent />
            </div>
        );
    }
}
