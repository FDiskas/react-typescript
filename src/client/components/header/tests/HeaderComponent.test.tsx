import * as React from 'react';
import { shallow } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';

import { HeaderComponent } from '../HeaderComponent';

describe('HeaderComponent.js', () => {
    it('renders without crashing', () => {
        shallow(<HeaderComponent/>);
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<HeaderComponent/>);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });
});
