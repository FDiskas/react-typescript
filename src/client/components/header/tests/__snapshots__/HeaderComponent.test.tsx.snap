// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent.js matches snapshot 1`] = `
<div
  className="header"
>
  Component name: HeaderComponent
  <NavigationComponent />
</div>
`;
